### PR TITLE
date_count_between translations

### PR DESCRIPTION
**Pull request by edward-burn**
For #1495, adds translations for clock::date_count_between() for a number of backends along with tests and update of news.

### CODE REVIEW

**Comment by edward-burn** [comment]
@hadley I noticed there was now a conflict with the news file so I have fixed that. Would it be possible to review this pr, hopefully it adds some nice functionality? (the matching functionality was also added in the duckdb 1.0 release https://github.com/duckdb/duckdb-r/pull/166)

**Comment by hadley** [comment]
I’m not working on dbplyr at the moment but I’ll definitely do it when I next am. (And I’ll take care of any merge conflicts so you don’t need to worry about that)

**Comment by edward-burn** [comment]
No problem @hadley, thanks for letting me know. When you do get to dbplyr, please just tag me if anything needs to be done and I'd be more than happy to contribute (I'll also be coming to the tidyverse developer day, so happy to work on anything related then too!)

**Comment by hadley** [comment]
@edward-burn thank you! And looking forward to seeing you at TDD!

File: R/backend-mssql.R [hunk 1]
```diff
@@ -372,6 +372,18 @@ simulate_mssql <- function(version = "15.0") {
       get_day = function(x) {
         sql_expr(DATEPART(DAY, !!x))
       },
+      date_count_between = function(start, end, precision, ..., n = 1L){
+
+        check_dots_empty()
+        if (precision != "day") {
+          cli::cli_abort('The only supported value for {.arg precision} on SQL backends is "day"')
```
Comment (simonpcouch) [inline R/backend-mssql.R:379]:
```suggestion
          cli_abort("{.arg precision} must be {.val day} on SQL backends.")
```

A couple notes here related to the [tidyverse style guide](https://style.tidyverse.org/errors.html) for error messages:

* `cli_abort()` is imported into the package namespace, so we don't need to namespace it!
* Using `.val` markup will allow us to use double-quotes.
* Added a period.
* Began the message with the problematic argument and used "must".

Could you apply these changes elsewhere in the PR? After those are made, I think this PR should be in a great spot.

File: NEWS.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 * `across(everything())` doesn't select grouping columns created via `.by` in
   `summarise()` (@mgirlich, #1493).
-  
+*  New translations of clock function `date_count_between()` for SQL server, Redshift, Snowflake, Postgres, and Spark (@edward-burn, #1495).
```
Comment (simonpcouch) [inline NEWS.md:5]:
```suggestion

*  New translations of clock function `date_count_between()` for SQL server, Redshift, Snowflake, Postgres, and Spark (@edward-burn, #1495).

```

**Review by simonpcouch — COMMENTED** [review]
Great! We'll be excited to get this merged in if you're game to revisit this PR for a few changes.

I've left one comment on how to revise an error message in this PR that should apply to each of the other new pieces of erroring code. Once the rest of those are resolved, this should be in a great spot!

**Comment by edward-burn** [comment]
thanks @simonpcouch! I have updated the pr based on the review, hopefully now all looks good

File: NEWS.md [hunk 3]
```diff
@@ -2,7 +2,9 @@
 
 * `across(everything())` doesn't select grouping columns created via `.by` in
   `summarise()` (@mgirlich, #1493).
-  
+
+*  New translations of clock function `date_count_between()` for SQL server, Redshift, Snowflake, Postgres, and Spark (@edward-burn, #1495).
```
Comment (simonpcouch) [inline NEWS.md:6]:
```suggestion
*  New translations of clock function `date_count_between()` for 
   SQL server, Redshift, Snowflake, Postgres, and Spark (@edward-burn, #1495).
```

**Review by simonpcouch — APPROVED** [review]
Thanks for making those changes, @edward-burn. Looking good!

**Pull request merged**